core/src/main/java/com/sequenceiq/cloudbreak/service/stackpatch/StackPatchUsageReporterService.java

```diff
@@ -0,0 +1,59 @@
+package com.sequenceiq.cloudbreak.service.stackpatch;
+
+import java.util.Objects;
+import java.util.Optional;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import com.cloudera.thunderhead.service.common.usage.UsageProto;
+import com.sequenceiq.cloudbreak.domain.stack.Stack;
+import com.sequenceiq.cloudbreak.domain.stack.StackPatchType;
+import com.sequenceiq.cloudbreak.usage.UsageReporter;
+
+@Service
+public class StackPatchUsageReporterService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(StackPatchUsageReporterService.class);
+
+    @Inject
+    private UsageReporter usageReporter;
+
+    public void reportAffected(Stack stack, StackPatchType stackPatchType) {
+        sendUsageReport(stack, stackPatchType, UsageProto.CDPStackPatchEventType.Value.AFFECTED);
+    }
+
+    public void reportSuccess(Stack stack, StackPatchType stackPatchType) {
+        sendUsageReport(stack, stackPatchType, UsageProto.CDPStackPatchEventType.Value.SUCCESS);
+    }
+
+    public void reportFailure(Stack stack, StackPatchType stackPatchType, String failureMessage) {
+        sendUsageReport(stack, stackPatchType, UsageProto.CDPStackPatchEventType.Value.FAILURE, failureMessage);
+    }
+
+    private void sendUsageReport(Stack stack, StackPatchType stackPatchType, UsageProto.CDPStackPatchEventType.Value eventType) {
+        sendUsageReport(stack, stackPatchType, eventType, "");
```

Comment:
Minor: instead of a hardcoded string literal I'd move prefer to simply pass null as optionalMessage - it is going to be empty string by `Objects.requireNonNullElse(optionalMessage, "")`